### PR TITLE
Refactor do immediate processing

### PR DESCRIPTION
**Pull request by Brett208**
The refactor didn't make the logic in OnImmediatePacketProcess as clean as I was hoping. reviewing the return statements and switch logic probably bears close scrutiny. I'm unsure if there are mistakes in the original code as the logic seems a bit too complicated.

----
EDIT: I based this on the branch IntegratePeerStatusEnum. I want to try switching the branch it is based on after merging the current base branch. I may need some help as this is a bit beyond my current comfort with Git.

### CODE REVIEW

**Comment by Brett208** [comment]
Well, I think that went badly?

File: client/OPUNetTransportLayer.cpp [hunk 1]
```diff
@@ -1076,26 +1076,9 @@ bool OPUNetTransportLayer::OnImmediatePacketProcess(Packet& packet, sockaddr_in&
...
-			// Send something to create router mappings
-			tlMessage.joinHelpRequest.clientAddr.sin_family = AF_INET;
-			if (tlMessage.joinHelpRequest.returnPortNum != 0) {
-				tlMessage.joinHelpRequest.clientAddr.sin_port = htons(tlMessage.joinHelpRequest.returnPortNum);
+			if (ProcessJoinHelpRequest(packet, fromAddress, tlMessage)) {
+				return true;
 			}
-			sendto(netSocket, (char*)&packet, 0, 0, (sockaddr*)&packet.tlMessage.joinHelpRequest.clientAddr, sizeof(packet.tlMessage.joinHelpRequest.clientAddr));
 
 			break;
```
Comment (DanRStevens) [inline client/OPUNetTransportLayer.cpp]: I appreciate how careful you are with these edits. I might not have thought to add the conditional.

**Review by DanRStevens — COMMENTED** [review]
Looks like you rebased on master, and then merged with the non-rebased history, rather than overwriting it.

**Comment by DanRStevens** [comment]
It may have been possible to not even do the rebase, now that the base branch is already merged to master.

----

At any rate, at this point, you can `git reset HEAD^` to remove the merge commit, run `git rebase master` to apply the branch changes on an updated master (make sure you update your local `master` branch first), and then `git push -f` to force push and overwrite the old history.

**Review by DanRStevens — APPROVED** [review]

**Comment by DanRStevens** [comment]
Oh, one comment I forgot to make earlier. In regards to #104, all of the listed method are event handlers methods, and so all can be named in the `OnX` style. It was only the root method with the integrated check that was maybe a little odd for that naming.

**Comment by Brett208** [comment]
Ah, thanks for fixing the merge issue. I was working coincident with you on the repository.

**Pull request merged**